### PR TITLE
adding orion into helm

### PR DESCRIPTION
**Pull request by joshuafernandes**
- workaround in place via configmap until orion is configurable via env vars

### CODE REVIEW

File: helm/private-network-ibft-and-orion/besu/values.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  validator3:
+    privateKey: "0x28021485044bf6870160f90b82ab6bbd14c691d4414140a63d047a7e3362c64e"
+    publicKey: "0x00b20ab6a385a2403d64637b3d93cb6d83215a08f29adb6feb4b8bf03387b734444e8b060f53150dea4b9b897823540d19918c13d6f57a5153d190b5fad7bf51"
+
+  validator4:
+    privateKey: "0xfb122a05ab1897ff144e1c9efb0bb3144f1e7f319aa5c55e20ef3d8d8464f4e8"
+    publicKey: "0x5fc1f8dc9f0c03087128e4bd724530e883d7de1a431269876dff9c95b8952f73c7e85ac7b49d85a2ad4950e967319482af435e07a0eab0a98d98449437787a00"
+
+orions:
```
Comment (arnaudgouriou) [inline helm/private-network-ibft-and-orion/besu/values.yaml]: Why only 2 key for the 4 orion nodes? With this configuration, it's mean that node 3 and 4 of orion have not load the public key in the `configmap.yaml`. It's what we want for this sample?

File: helm/private-network-ibft/besu/templates/validator2-statefulset.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ spec:
       app: {{ template "besu.name" . }}
       release: {{ .Release.Name }}
       component: validator2
+  serviceName: {{ template "besu.fullname" . }}-validator2
```
Comment (arnaudgouriou) [inline helm/private-network-ibft/besu/templates/validator2-statefulset.yaml]: same as above about create a variable for the name `validator2`

File: kubectl/private-network-ibft-and-orion/README.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+![Image ibft](../../images/ibft-orion.png)
+
+## NOTE:
+1. validators1 and 2 serve as bootnodes as well. Adjust according to your needs
+2. If you add more validators in past the initial setup, they need to be voted in to be validators i.e they will serve as normal nodes and not validators until they've been voted in.
+3. Node1Privacy to Node4Privacy are tied to Orion1 to Orion4 respectively
+
+#### 1. Boot nodes private keys
+Create private/public keys for the validators using the besu subcommands. The private keys are put into secrets and the public keys go into a configmap to get the bootnode enode address easily
+Repeat this process for as many validators as you would like to provision i.e keys and replicate the deployment & service
```
Comment (arnaudgouriou) [inline kubectl/private-network-ibft-and-orion/README.md]: same as above about the setting of number of key you can generate

File: helm/private-network-ibft-and-orion/besu/README.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+For more information please refer to the [documentation](https://docs.orion.pegasys.tech/en/stable/Getting-Started/Quickstart/#2-generate-keys) 
+Create the keypairs and enter the password when requested. 
+```bash
+docker run -it --volume $PWD/orionSetup/orion1:/opt/orion/data --entrypoint "/bin/sh" pegasyseng/orion:develop -c 'cd /opt/orion/data && /opt/orion/bin/orion -g nodeKey'
+docker run -it --volume $PWD/orionSetup/orion2:/opt/orion/data --entrypoint "/bin/sh" pegasyseng/orion:develop -c 'cd /opt/orion/data && /opt/orion/bin/orion -g nodeKey' 
+sudo chown -R $USER:$USER ./orionSetup
+```
+
+#### 4. Orion configuration
+Update the orion1.conf & orion2.conf files to suit requirements
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/README.md]: :+1:

File: helm/private-network-ibft-and-orion/besu/orion1.key [hunk 5]
```diff
@@ -0,0 +1 @@
+{"data":{"bytes":"hBsuQsGJzx4QHmFmBkNoI7YGnTmaZP4P+wBOdu56ljk="},"type":"unlocked"}
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/orion1.key]: This one is without password. I'll update to use a password in line. Orion needs a bit of work with config (& doco)

File: helm/private-network-ibft-and-orion/besu/values.yaml (hunk 1 above)
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/values.yaml]: Yup, fixed

File: helm/private-network-ibft-and-orion/besu/genesis.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  "nonce" : "0x0",
+  "timestamp" : "0x58ee40ba",
+  "gasLimit" : "0x47b760",
+  "difficulty" : "0x1",
+  "mixHash" : "0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365",
+  "coinbase" : "0x0000000000000000000000000000000000000000",
+  "alloc" : {
+    "fe3b557e8fb62b89f4916b721be55ceb828dbd73" : {
+      "privateKey" : "8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63",
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/genesis.json]: Dont need to to be honest, kept it to show people where to add & its the same as the doco. Realistically we're expecting people to change it and use their own

File: helm/private-network-ibft-and-orion/besu/ibftSetup/ibftConfigFile.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    "nonce": "0x0",
+    "timestamp": "0x58ee40ba",
+    "gasLimit": "0x47b760",
+    "difficulty": "0x1",
+    "mixHash": "0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365",
+    "coinbase": "0x0000000000000000000000000000000000000000",
+    "alloc": {
+      "fe3b557e8fb62b89f4916b721be55ceb828dbd73": {
+        "privateKey": "8f2a55949038a9610f50fb23b5883af3b4ecb3c3bb792cbcefbd1542c692be63",
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/ibftSetup/ibftConfigFile.json]: Dont need to to be honest, kept it to show people where to add & its the same as the doco. Realistically we're expecting people to change it and use their own

File: helm/private-network-ibft-and-orion/besu/templates/node-statefulset.yaml [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        app: {{ template "besu.name" . }}
+        release: {{ .Release.Name }}
+        component: node
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "9545"
+        prometheus.io/path: "/metrics"
+    spec:
+      initContainers:
+        - name: init-bootnode2
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/templates/node-statefulset.yaml]: :+1:

File: helm/private-network-ethash/besu/templates/bootnode2-statefulset.yaml [hunk 9]
```diff
@@ -16,6 +16,7 @@ spec:
       app: {{ template "besu.name" . }}
       release: {{ .Release.Name }}
       component: bootnode2
+  serviceName: {{ template "besu.fullname" . }}-bootnode1
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ethash/besu/templates/bootnode2-statefulset.yaml]: :+1: REplicated all across now

File: helm/private-network-ibft-and-orion/besu/README.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+## NOTE:
+1. validators1 and 2 serve as bootnodes as well. Adjust according to your needs
+2. If you add more validators in past the initial setup, they need to be voted in to be validators i.e they will serve as normal nodes and not validators until they've been voted in.
+3. Node1Privacy to Node4Privacy are tied to Orion1 to Orion4 respectively
+
+## Pre chart install - you need to create config that you want to persist
+#### 1. Validators private keys
+Create private/public keys for the validators using the besu subcommands. The private keys are put into secrets and the public keys go into a configmap to get the bootnode enode address easily
+Repeat this process for as many validators as you would like to provision i.e keys and replicate the deployment & service
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/README.md]: Good catch :+1:

File: helm/private-network-ethash/besu/templates/bootnode2-statefulset.yaml [hunk 11]
```diff
@@ -109,6 +113,10 @@ spec:
                 --rpc-ws-enabled \
                 --rpc-ws-host=0.0.0.0 \
                 --rpc-ws-port=8546 \
+                --graphql-http-enabled \
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ethash/besu/templates/bootnode2-statefulset.yaml]: Couldnt get the env var to working without helm complaining about template errors, have done this instead though `{{- if .Values.nodeInterfaces.rpcEnabled }}`

File: helm/private-network-ibft-and-orion/besu/orion1.key (hunk 5 above)
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/orion1.key]: I used the command in step 3 of the Readme. We should do something with password so I'll leave your other comment up and resolve this one

File: helm/private-network-ibft-and-orion/besu/templates/configmap.yaml [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+  orion4PubKey: {{ trimPrefix "0x" .Values.orions.orion4.publicKey | quote }}
+  {{ else }}
+  orion4PubKey: {{ .Values.orions.orion4.publicKey | quote }}
+  {{ end }}
+
+
+  genesis.json: |-
+{{ .Files.Get "genesis.json" | indent 4}}
+
+  orion1.conf: |-
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/templates/configmap.yaml]: :+1:

File: helm/private-network-ibft-and-orion/besu/templates/node-statefulset.yaml [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+              protocol: TCP
+            - containerPort: 30303
+              name: dev-p2p-udp
+              protocol: UDP
+          command:
+            - /bin/sh
+            - -c
+          args:
+            - |
+              exec /opt/besu/bin/besu \
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/templates/node-statefulset.yaml]: Good point :+1: I'll update

File: helm/private-network-ibft-and-orion/besu/templates/orions-keys-secret.yaml [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+  name: {{ template "besu.fullname" . }}-orion-keys
+  labels:
+    app: {{ template "besu.name" . }}
+    chart: {{ template "besu.chart" . }}
+    release: {{ .Release.Name }}
+    namespace: {{ .Release.Namespace }}
+    heritage: {{ .Release.Service }}
+type: Opaque
+data:
+{{ (.Files.Glob "orion1.key").AsSecrets | indent 2 }}
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/templates/orions-keys-secret.yaml]: Ease of use for the moment, but did want to have a chat with you about RBACs for each secret mapped to specific nodes esp in light of the official helm chart we release

File: helm/private-network-ethash/besu/templates/bootnode2-statefulset.yaml (hunk 11 above)
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ethash/besu/templates/bootnode2-statefulset.yaml]: I think the config file via configmap is the best long term approach. I'll switch to that instead

File: helm/private-network-ibft-and-orion/besu/templates/orions-keys-secret.yaml (hunk 14 above)
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/templates/orions-keys-secret.yaml]: Seperated files now with passwords

File: helm/private-network-ibft-and-orion/besu/README.md [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+cp ./ibftSetup/networkFiles/genesis.json ./
+```
+
+Update the values.yaml with the keys. The private keys are put into secrets and the public keys go into a configmap that other nodes use to create the enode address
+
+#### 2. Genesis.json
+The genesis.json file generated should have been placed at the root directory of this helm chart
+
+#### 3. Orion keys
+For more information please refer to the [documentation](https://docs.orion.pegasys.tech/en/stable/Getting-Started/Quickstart/#2-generate-keys)
```
[the remark this answers is not in the document]
Comment (joshuafernandes) [reply helm/private-network-ibft-and-orion/besu/README.md]: I've updated to with passwords for Orion, readme has steps I used pt4

**Pull request merged**